### ISSUE TITLE
upload file failed

### ISSUE THREAD

**Issue by ljwps** [langgenius/dify#10845]
### Self Checks

- [X] This is only for bug report, if you would like to ask a question, please head to [Discussions](https://github.com/langgenius/dify/discussions/categories/general).
- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### Dify version

0.11.1

### Cloud or Self Hosted

Self Hosted (Docker)

### Steps to reproduce

**Upgrade from 0.6.x to 0.11.1**

**Upgrade Process:**

1. The local version was originally 0.6.x from Git.
2. Switched to the 0.11.1 tag.
3. Ran docker-compose down.
4. Ran docker-compose pull.
5. Ran docker-compose up -d.

After starting up, I created a node with a file upload feature for testing. When trying to upload a .docx file in the preview mode, an error appeared saying "File type not supported." I also tried uploading an .xlsx file and encountered the same error. However, uploading an .xml file was successful.

Using the browser's F12 developer tools to inspect the file upload request, I noticed that no request was actually sent. This suggests that the file type check is performed on the client side, as there are no error logs on the server.

**Screenshots** 
![image](https://github.com/user-attachments/assets/c698b0f4-aa43-4991-86ee-1f2c3df3b27e)

![image](https://github.com/user-attachments/assets/c07bb96d-333f-4f98-9fa5-54726eea95d0)

![image](https://github.com/user-attachments/assets/ece5028d-4e06-4464-aa88-dd4b6adecdbc)

![image](https://github.com/user-attachments/assets/9be22f98-c085-4fba-acfe-bfc56c6f0bd3)

![image](https://github.com/user-attachments/assets/6612e6a5-7652-4329-a510-3480a3fb30c2)

### ✔️ Expected Behavior

upload file success

### ❌ Actual Behavior

File type not supported

**Comment by ljwps** [comment]
Today, upgraded to version 0.11.2 following the same process, but the issue persists.

**Comment by hanqingwu** [comment]
yes, it  persists at 0.11.2 . I think root cause in web frontend . because i don't catch web send request api.

**Comment by sigvardt** [comment]
Also experincing it when I use the workflow API.

**Comment by crazywoola** [comment]
Can you provide an example doc here. I can not reproduce this problem with this file type.

**Comment by sigvardt** [comment]
@crazywoola more details here in a similar issue I opened: 

https://github.com/langgenius/dify/issues/10557

**Comment by perzeuss** [comment]
@crazywoola I am able to reproduce it. The file upload works, but as soon as you run the chatflow with the uploaded file in the request (upload_file_id), you get a 400 response.
In was able to reproduce it on both console api (https://cloud.dify.ai/console/api/apps/.../advanced-chat/workflows/draft/run) and webapp api (https://udify.app/api/chat-messages)
```json
{"code": "invalid_param", "message": "Invalid upload file", "status": 400}
```

Example file: [Random Reviews.xlsx](https://github.com/user-attachments/files/17906694/Random.Reviews.xlsx)

Screenshot: ![image](https://github.com/user-attachments/assets/409678b5-45d8-45b0-bd7a-b473911bb113)


Example workflow:
```yaml
app:
  description: ''
  icon: 🤖
  icon_background: '#FFEAD5'
  mode: advanced-chat
  name: XLSX Document Chatflow
  use_icon_as_answer_icon: false
kind: app
version: 0.1.3
workflow:
  conversation_variables: []
  environment_variables: []
  features:
    file_upload:
      allowed_file_extensions:
      - .JPG
      - .JPEG
      - .PNG
      - .GIF
      - .WEBP
      - .SVG
      allowed_file_types:
      - image
      allowed_file_upload_methods:
      - local_file
      - remote_url
      enabled: false
      fileUploadConfig:
        audio_file_size_limit: 50
        batch_count_limit: 5
        file_size_limit: 15
        image_file_size_limit: 10
        video_file_size_limit: 100
        workflow_file_upload_limit: 10
      image:
        enabled: false
        number_limits: 3
        transfer_methods:
        - local_file
        - remote_url
      number_limits: 3
    opening_statement: ''
    retriever_resource:
      enabled: true
    sensitive_word_avoidance:
      enabled: false
    speech_to_text:
      enabled: false
    suggested_questions: []
    suggested_questions_after_answer:
      enabled: false
    text_to_speech:
      enabled: false
      language: ''
      voice: ''
  graph:
    edges:
    - data:
        isInIteration: false
        sourceType: start
        targetType: answer
      id: 1732553887444-source-answer-target
      source: '1732553887444'
      sourceHandle: source
      target: answer
      targetHandle: target
      type: custom
      zIndex: 0
    nodes:
    - data:
        desc: ''
        selected: false
        title: Start
        type: start
        variables:
        - allowed_file_extensions: []
          allowed_file_types:
          - document
          allowed_file_upload_methods:
          - local_file
          - remote_url
          label: document
          max_length: 48
          options: []
          required: true
          type: file
          variable: document
      height: 90
      id: '1732553887444'
      position:
        x: 30
        y: 250
      positionAbsolute:
        x: 30
        y: 250
      selected: true
      sourcePosition: right
      targetPosition: left
      type: custom
      width: 244
    - data:
        answer: Just testing the upload
        desc: ''
        selected: false
        title: Answer
        type: answer
        variables: []
      height: 100
      id: answer
      position:
        x: 334
        y: 250
      positionAbsolute:
        x: 334
        y: 250
      selected: false
      sourcePosition: right
      targetPosition: left
      type: custom
      width: 244
    viewport:
      x: 341
      y: 35
      zoom: 0.7
```

**Comment by zolgear** [comment]
After upgrading to 0.12.0, the error "Invalid upload file" reported by @perzeuss occurred.  

- File upload is confirmed to be successful, and files are generated under `app/storage/upload_files/...`.  

Since the upload itself is successful and the issue appeared in 0.12.0, it seems to be unrelated to the original issue.

**Comment by crazywoola** [comment]
We have several fixes related to #11112 #11111 #11114 @zolgear @perzeuss Please wait for the 0.12.1

**Comment by lzcmagic** [comment]
> We have several fixes related to #11112 #11111 #11114 @zolgear @perzeuss Please wait for the 0.12.1

This version does not report 'upload file failed', and the file variable is also not functioning properly。

**Comment by ChungRae** [comment]
The parameter passed by the front end is wrong and is always the default option
![微信图片_20241212145557](https://github.com/user-attachments/assets/e323886b-b15a-4881-b3ff-c05b47613642)
![微信图片_2024-12-12_145730_439](https://github.com/user-attachments/assets/6a9b2259-f1f1-4b0a-9afe-e97c14635c49)
